pandas/core/groupby/ops.py

```diff
@@ -95,6 +96,64 @@
     get_indexer_dict,
 )
 
+_CYTHON_FUNCTIONS = {
+    "aggregate": {
+        "add": "group_add",
+        "prod": "group_prod",
+        "min": "group_min",
+        "max": "group_max",
+        "mean": "group_mean",
+        "median": "group_median",
+        "var": "group_var",
+        "first": "group_nth",
+        "last": "group_last",
+        "ohlc": "group_ohlc",
+    },
+    "transform": {
+        "cumprod": "group_cumprod",
+        "cumsum": "group_cumsum",
+        "cummin": "group_cummin",
+        "cummax": "group_cummax",
+        "rank": "group_rank",
+    },
+}
+
+
+@functools.lru_cache(maxsize=None)
+def _get_cython_function(kind: str, how: str, dtype: np.dtype, is_numeric: bool):
```

Comment:
This function is a straight cut&paste of the existing method. But moving it to a standalone function instead of method to be able to use lru_cache on it.